tests/conftest.py

```
@@ -185,6 +185,67 @@ def registry_host(pytestconfig: pytest.Config) -> list[str]:
+    authorization = (
+        f"AWS4-HMAC-SHA256 Credential={aws_access_key_id}/{credential_scope}, "
+        f"SignedHeaders=content-type;host;x-amz-date, Signature={signature}"
+    )
+
+    req = Request(
+        f"https://{host}/",
+        data=body.encode(),
+        headers={"Content-Type": content_type, "X-Amz-Date": amzdate, "Authorization": authorization},
+    )
+    try:
+        urlopen(req, timeout=10)
```

Ruff (0.15.9)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)